src/components/OwnerPanel/OwnerPanel.tsx

```diff
@@ -63,7 +63,8 @@ class OwnerPanel extends React.Component<Props, State> {
                 style={{
                   width: `${
                     (this.props.amountRaised / this.props.targetAmount) * 100
-                  }%`,
+                    }%`,
+                  backgroundColor: `${this.props.progressBarColor ? this.props.progressBarColor : ''}`
```

Comment:
What happens if the background color is `''`? Can we also do some validation that what is passed from the server is of form #XXXXXX` aka is a html hex color code?

Reply:
Mind adding a comment here that when the backgroundColor is `''` it defaults to what's defined in the CSS?

Reply:
I could do a regex pattern check if its a valid html hex and if it doesn’t pass, i can set it to our default color pattern.

Reply:
That would be perfect!

Reply:
It would be great if we had a util or or factory of source that all of the data from the server passed through so that all of the sanitization of server data could be put in one place. Plus it would be easy to unit test.

If you do that, it could be a follow-up. Just be sure to create a ticket on Trello so that we can track it.

Reply:
i see. since its nonblocking i simplified it and added a comment. as mentioned its not validated yet so if someone enters an invalid hex in defaults to gray. if its null it defaults to the color that was assigned in the css file

Reply:
@jt5301 should i add the regex on top of your code to validate the color or are you handling the whole progress bar

Reply:
@bvillaroman Feel free to add! I haven’t really taken ownership of the whole thing yet, was just doing color for now

Reply:
Let's merge this and add a separate ticket for sanitization

Reply:
Anyone feel free to pick it up and add it into the sprint: https://trello.com/c/l1bzIYjg/162-add-a-util-to-sanitize-server-responses